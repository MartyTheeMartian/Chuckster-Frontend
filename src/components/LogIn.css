.login-ctn {
  min-width: 240px;
}
