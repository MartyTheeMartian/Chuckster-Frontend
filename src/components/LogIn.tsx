import { useRef, useContext } from 'react';
import { UserContext } from 'context/UserContext';
import { print } from 'graphql';
import Cookies from 'js-cookie';
import axios from 'axios';
import SignUp from 'components/SignUp';
import config from 'config/api';
import userAuth from 'gql/mutations/userAuth.gql';
import { JWT_TOKEN } from 'assets/constants';
import './LogIn.css';


export default function LogIn(): JSX.Element {

  const { user, setUser } = useContext(UserContext);

  const emailInputRef = useRef<HTMLInputElement>(null);
  const passwordInputRef = useRef<HTMLInputElement>(null);

  const fetchLogIn = async (email: string, password: string): Promise<void> => {
    try {
      const { data } = await axios.post(config.baseURL, {
        query: print(userAuth),
        variables: {
          email,
          password,
        },
      },
      config.getHeaders()
      );

      const jwtToken = JSON.parse(data.data.authenticate.jwtToken);

      if (jwtToken) {
        Cookies.set(JWT_TOKEN, jwtToken);

        setUser({
          ...user,
          authenticated: true,
          email,
          password,
        });
      } else {
        Cookies.remove(JWT_TOKEN);

        setUser({
          ...user,
          authenticated: false,
        });
      }
    } catch (error) {
      console.error(error);
    }
  };

  const submitClick = (event: React.MouseEvent<HTMLElement>): void => {
    event.preventDefault();

    const emailInput = emailInputRef.current?.value ?? '';
    const passwordInput = passwordInputRef.current?.value ?? '';

    fetchLogIn(emailInput, passwordInput).catch(error => console.error(error));
  };

  return (
    <div className='login-ctn'>
      <h2>
        Login
      </h2>
      <div>
        <form>
          <input
            ref={emailInputRef}
            type='email'
            placeholder='email' />
          <input
            ref={passwordInputRef}
            type='password'
            placeholder='password' />
          <button
            type='submit'
            onClick={submitClick}>
            Submit
          </button>
        </form>
      </div>
      <SignUp />
    </div>
  );
}
