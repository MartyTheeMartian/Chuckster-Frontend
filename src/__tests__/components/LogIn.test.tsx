import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LogIn from 'components/LogIn';

it('renders App component', () => {
  const result = shallow(<LogIn />);
  expect(toJson(result)).toMatchSnapshot();
});
