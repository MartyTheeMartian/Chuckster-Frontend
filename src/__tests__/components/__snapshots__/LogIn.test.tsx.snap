// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div
  className="login-ctn"
>
  <h2>
    Login
  </h2>
  <div>
    <form>
      <input
        placeholder="email"
        type="email"
      />
      <input
        placeholder="password"
        type="password"
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <SignUp />
</div>
`;
